From 1ad3b1e3256a226be362de1a4959f2a642d349b7 Mon Sep 17 00:00:00 2001
From: Peter Qiu <zqiu@google.com>
Date: Mon, 30 Jan 2017 12:55:09 -0800
Subject: WifiInstaller: add permission for access downloaded files

The previous fix for the security vulnerability b/33178389
add the code to drop the downloaded installation file but
forget to update the manifest for additional permission.  This
omission was due to that the original fix was merged from
mnc-dev, which doesn't have such permission requirement.

So add the required permission for deleting downloaded
installation file.

Bug: 33178389
Test: Verify WifiInstaller works and not crashing due permission
      denied exception

Change-Id: I987ad73dde0e3cc5b920cd08830dd065c47be402
(cherry picked from commit 9a9e51727a5cdaf3ac10c98e1ea4661bc13a11c7)
---
 AndroidManifest.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index cc02a95..f6ad323 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -7,6 +7,7 @@
                 android:protectionLevel="signature" />
     <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_ALL_DOWNLOADS" />
 
     <application android:label="@string/app_name"
                  android:allowBackup="false">
-- 
cgit v1.1

